### PR TITLE
[Win32] Add build notes for win-win32

### PR DESCRIPTION
**Pull request by TyHolc**
b/281735840

Change-Id: Ic1a8b64eba9fdf80c1e150d478418f89ba7ff4d0

### CODE REVIEW

File: cobalt/site/docs/development/setup-windows.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+1.  Compile the code from the `cobalt/` directory:
+
+    ```
+    $ ninja -C out/<platform>_<build_type> <target_name>
+    ```
+
+    The previous command contains three variables:
+
+    1.  `<platform>` is the [platform
+        configuration](/starboard/porting.html#1-enumerate-and-name-your-platform-configurations)
```
[the remark this answers is not in the document]
Comment (TyHolc) [reply cobalt/site/docs/development/setup-windows.md:141]: This should resolve to the relative path once it's hosted. I don't think github is aware of that link, but I'll double check this once this page is up.

File: cobalt/site/docs/development/setup-windows.md [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+<aside class="note">
+  <b>Note:</b> if you change the value of `PUBLISHER` in
+  `appx_product_settings.py` you <b>must</b> regenerate a pfx file in order for
+  the packaging step below to work correctly. Follow the instructions in
+  `starboard/xb1/cert/README.md` to generate your own pfx.
+</aside>
+
+### Build Cobalt
+
+To build Cobalt for the Xbox One, set the platform to `xb` in the gn step:
```
[the remark this answers is not in the document]
Comment (TyHolc) [reply cobalt/site/docs/development/setup-windows.md:232]: Yes, good catch!

File: cobalt/site/docs/development/setup-windows.md [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+
+1.  Install GN, which we use for our build system code. There are a few ways to
+    get the binary, follow the instructions for whichever way you prefer
+    [here](https://cobalt.googlesource.com/third_party/gn/+/refs/heads/main/#getting-a-binary).
+
+1.  (Optional)
+    [Install Sccache](https://github.com/mozilla/sccache#installation) to
+    support build acceleration
+
+1.  Make sure the all of the above installed packages are on your Path environment
```
[the remark this answers is not in the document]
Comment (TyHolc) [reply cobalt/site/docs/development/setup-windows.md:44]: Done.

File: cobalt/site/docs/development/setup-windows.md [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+      * Microsoft.VisualStudio.Component.VC.Llvm.Clang
+      * Microsoft.VisualStudio.Component.VC.Llvm.ClangToolset
+      * Microsoft.VisualStudio.Component.Windows10SDK.18362
+      * Microsoft.VisualStudio.Workload.NativeDesktop
+    * [winflexbison](https://github.com/lexxmark/winflexbison)
+
+    <aside class="note">
+      <b>Note:</b> By default, Cobalt's build system will check
+      C:\Program Files (x86)\ for the Visual Studio install directory. If you
+      installed it elsewhere, you can set the `VSINSTALLDIR` environment
```
[the remark this answers is not in the document]
Comment (TyHolc) [reply cobalt/site/docs/development/setup-windows.md:31]: It's appearing correctly in g3docs, but I'm not sure which one cobalt.dev will follow. I'll leave them as-is for now and check in on it once the page is up, updating if needed.

**Review by arjungm — APPROVED** [review]

**Review by arjungm — APPROVED** [review]

File: cobalt/site/docs/development/setup-windows.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+      `C:/Program Files/Microsoft Visual Studio/2022/Professional`
+    </aside>
+
+1.  Install GN, which we use for our build system code. There are a few ways to
+    get the binary, follow the instructions for whichever way you prefer
+    [here](https://cobalt.googlesource.com/third_party/gn/+/refs/heads/main/#getting-a-binary).
+
+1.  (Optional)
+    [Install Sccache](https://github.com/mozilla/sccache#installation) to
+    support build acceleration
```
Comment (alexanderahn) [inline cobalt/site/docs/development/setup-windows.md:42]: nit: missing period

**Review by alexanderahn — APPROVED** [review]

File: cobalt/site/docs/development/setup-windows.md (hunk 5 above)
Comment (TyHolc) [reply cobalt/site/docs/development/setup-windows.md]: Done.

**Comment by kaidokert** [comment]
Just a ping for @oxve as we want to get this indexed in https://cobalt.dev eventually.

**Review by kaidokert — APPROVED** [review]

**Comment by TyHolc** [comment]
Looks like the check bug ID is failing because I synced my fork and it created a merge-commit that doesn't have a bug ID (woops), despite all of **my** commits having them. What's the easiest way around this?

**Comment by kaidokert** [comment]
> Looks like the check bug ID is failing because I synced my fork and it created a merge-commit that doesn't have a bug ID (woops), despite all of **my** commits having them. What's the easiest way around this?

The Bug ID is now _only_ needed in PR description - the part thats editable here in Github UI. What's in commit messages themselves doesn't matter anymore. Looks like the check is green now too

**Comment by oxve** [comment]
If you want this up on cobalt.dev just cherry pick to 24.lts and let me know when it's merged.

**Pull request merged**